app/src/App.vue

```diff
@@ -9,24 +9,43 @@ import { getSocket } from "luminary-shared";
 import { apiUrl, initLanguage } from "./globalConfig";
 import NotificationManager from "./components/notifications/NotificationManager.vue";
 
-const { isAuthenticated, getAccessTokenSilently, loginWithRedirect } = useAuth0();
+const { isAuthenticated, getAccessTokenSilently, loginWithRedirect, logout } = useAuth0();
 
 initLanguage();
 
+const loginRedirect = async () => {
+    const usedConnection = localStorage.getItem("usedAuth0Connection");
+    const retryCount = parseInt(localStorage.getItem("auth0AuthFailedRetryCount") || "0");
+
+    // Try to login. If this fails (e.g. the user cancels the login), log the user out after the second attempt
+    if (retryCount < 2) {
+        localStorage.setItem("auth0AuthFailedRetryCount", (retryCount + 1).toString());
+        await loginWithRedirect({
+            authorizationParams: {
+                connection: usedConnection ? usedConnection : undefined,
+                redirect_uri: window.location.origin,
+            },
+        });
+        return;
+    }
+
+    localStorage.removeItem("auth0AuthFailedRetryCount");
+    localStorage.removeItem("usedAuth0Connection");
+    await logout({ logoutParams: { returnTo: window.location.origin } });
+};
+
+// Clear the auth0AuthFailedRetryCount if the user logs in successfully (if the app is not redirecting to the login page, we assume the user either logged out or the login was successful)
+setTimeout(() => {
+    localStorage.removeItem("auth0AuthFailedRetryCount");
+}, 10000);
```

Comment:
Doesn't `loginWithRedirect` always redirect to the login page? If you're already logged in you will immediately be redirected back without actually seeing the page, but it will route you via Auth0 I think. I feel like the clearing of this rather belongs in the `onBeforeMount` where we have the `await waitUntilAuth0IsLoaded()`, after which we can be sure Auth0 is initialized. But perhaps you've already thought of all of this and there is a reason this is here. As you know I always itch when I see a `setTimeout` 😛

Reply:
No, I tested, and it does not always go to the login page. If you still have a valid session, it redirects you back to the app with a working token.

Do not like the timeout either, but I cannot think of another way to determine if the user is logged into the app successfully to clear the retry counter.